File: competitive_programming/leetcode/06_javascript/283_move_zeroes.js
function moveZeroes(nums) {
  const result = nums.filter((number) => number !== 0);
  while (result.length < nums.length) result.push(0);
  return result;
}

if (require.main === module) {
  if (moveZeroes([0, 1, 0, 3, 12]).join(",") !== "1,3,12,0,0") throw new Error("move zeroes failed");
  if (moveZeroes([0]).join(",") !== "0") throw new Error("move zeroes failed");
  console.log("283 move zeroes ok");
}

File: competitive_programming/leetcode/11_golang/283_move_zeroes.go
package main

import "fmt"

func moveZeroes(nums []int) []int {
	result := []int{}
	for _, number := range nums {
		if number != 0 {
			result = append(result, number)
		}
	}
	for len(result) < len(nums) {
		result = append(result, 0)
	}
	return result
}

func main() {
	if fmt.Sprint(moveZeroes([]int{0, 1, 0, 3, 12})) != "[1 3 12 0 0]" || fmt.Sprint(moveZeroes([]int{0})) != "[0]" {
		panic("move zeroes failed")
	}
	fmt.Println("283 move zeroes ok")
}
